tests/test_admin.py

```diff
@@ -0,0 +1,167 @@
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.get(
+                self.get_admin_url(PageContent, "duplicate_content", pagecontent.pk)
+            )
+        self.assertEqual(response.status_code, 200)
+
+    def test_post_empty_slug(self):
+        pagecontent = PageContentWithVersionFactory()
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.post(
+                self.get_admin_url(PageContent, "duplicate_content", pagecontent.pk),
+                data={"slug": ""},
+            )
+            form = response.context["form"]
+        self.assertEqual(response.status_code, 200)
+        self.assertFalse(form.is_valid())
+        self.assertIn("slug", form.errors)
+        self.assertEqual(form.errors["slug"], ["This field is required."])
+
+    def test_post_empty_slug_after_slugify(self):
+        pagecontent = PageContentWithVersionFactory()
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.post(
+                self.get_admin_url(PageContent, "duplicate_content", pagecontent.pk),
+                data={"site": Site.objects.first().pk, "slug": "£"},
+            )
+            form = response.context["form"]
+        self.assertEqual(response.status_code, 200)
+        self.assertFalse(form.is_valid())
+        self.assertIn("slug", form.errors)
+        self.assertEqual(form.errors["slug"], ["Slug must not be empty."])
+
+    def test_post(self):
+        pagecontent = PageContentWithVersionFactory(template="page.html")
+        placeholder = PlaceholderFactory(slot="content", source=pagecontent)
+        add_plugin(placeholder, "TextPlugin", pagecontent.language, body="Test text")
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.post(
+                self.get_admin_url(PageContent, "duplicate_content", pagecontent.pk),
+                data={"site": Site.objects.first().pk, "slug": "foo"},
+                follow=True,
+            )
+        self.assertRedirects(response, self.get_admin_url(PageContent, "changelist"))
+        new_pagecontent = PageContent._base_manager.latest("pk")
+        new_placeholder = new_pagecontent.placeholders.get(slot="content")
+        self.assertEqual(PageContent._base_manager.count(), 2)
+        self.assertNotEqual(pagecontent, new_pagecontent)
+        self.assertNotEqual(pagecontent.page, new_pagecontent.page)
+        self.assertEqual(pagecontent.language, new_pagecontent.language)
+        self.assertEqual(new_pagecontent.page.get_slug(new_pagecontent.language), "foo")
+        new_plugins = list(downcast_plugins(new_placeholder.get_plugins_list()))
+        self.assertEqual(len(new_plugins), 1)
+        self.assertEqual(new_plugins[0].plugin_type, "TextPlugin")
+        self.assertEqual(new_plugins[0].body, "Test text")
+
+
+class RegistrationTestCase(CMSTestCase):
```

Comment:
I think better to use `CMSTestCase` when you genuinely need it. Otherwise stick to django's `TestCase` which is less likely to create much unnecessary overhead/slow down tests.